app/models/uploaded_file.rb

```diff
@@ -5,97 +5,30 @@ class UploadedFile < ActiveRecord::Base
             presence: true,
             inclusion: %w(cita_appointments)
   validates :filename,
-            presence: true,
             format: /\A.*\.csv\z/
+  validates :data,
+            presence: true
 
   validate :correct_headers
 
-  scope :pending, -> { where(processed: false) }
+  scope :pending, -> { where(processed: false).order('created_at DESC') }
 
   def correct_headers
+    return unless data.present?
+
     headers = CSV.new(StringIO.new(data)).first
-    errors.add(:data, :headers) unless headers == HEADERS
+    missing_headers = REQUIRED_HEADERS - headers
+
+    errors.add(:data, :headers, missing: missing_headers.join(', ')) unless missing_headers.empty?
```

Comment:
Maybe `missing_headers.to_sentence` would produce a nicer output and be a little clearer / more concise?